### PR TITLE
Move front end source files to web_src and generate index.js minify version

### LINKED ISSUE #8430: Minify Javascript
Javascript minify could be better in gitea. According to this test https://gtmetrix.com/reports/try.gitea.io/4Rz4MxWz it would reduce the size by 48KiB.

### PR DESCRIPTION
**Pull request by lunny**
Fix #8430 and this will also reduce gitea package size since we moved frontend source files from public directory.

### CODE REVIEW

**Comment by guillep2k** [comment]
Was `npm` already a requirement for building from source?

**Comment by guillep2k** [comment]
Shouldn't we lock to a specific version of the minifier? Otherwise CI build might differ from the committed version.

File: Makefile [hunk 1]
```diff
@@ -418,6 +418,10 @@ npm-check:
 		exit 1; \
 	fi;
 
+.PHONY: webpack
+webpack: npm
+	npm install webpack -g
+
```
Comment (silverwind) [inline Makefile:424]: This is unnecessary, just run `npm install --save-dev webpack` which will add it as a proper dependency. To run it, use `npx webpack <args>` afterwards.

**Comment by silverwind** [comment]
This doesn't really solve https://github.com/go-gitea/gitea/issues/8430. All that webpack config does is copy source to destination. If you want to minify, you would at least want to add a minify plugin, for which I recommend `terser`:

https://github.com/webpack-contrib/terser-webpack-plugin

**Review by gary-kim — CHANGES_REQUESTED** [review]
We should add a `package.json` and handle dependencies there. We also need to set Webpack to production mode for the final compile, either in the Webpack config or in the command for it.

**Comment by lunny** [comment]
@gary-kim we already have a `package.json` file on home dir.

**Comment by gary-kim** [comment]
🤦 Forgot about LESS.
What @silverwind said then.

**Pull request closed without merging**